emails/models.py

```diff
@@ -270,7 +293,18 @@ def __init__(self, message=None):
         self.message = message
 
 
+class DomainPickingManager(models.Manager):
+    def create(self, *args, **kwargs):
+        if 'domain' not in kwargs:
+            kwargs['domain'] = get_domain_from_env_vars_and_profile(
+                kwargs['user']
+            )
+        return super().create(*args, **kwargs)
+
+
```

Comment:
Moving all the logic into a single place (either in `create` here or in `save` for `RelayAddress`. I would recommend moving the logic all in `save` since it's safer there.